sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/IndexAction.java

```diff
@@ -29,7 +29,7 @@
     private Map<String, Object> properties;
 
     @JsonAnyGetter
-    private Map<String, Object> getParamMap() {
+    public Map<String, Object> getParamMap() {
```

Comment:
Did you mean to make this public again?

Reply:
For indexAction, we can define either a map (properties) or an object(document).
If we made this private, we cannot access properties only document.

Reply:
Could this be package private instead?

Reply:
It is currently not used by any API. However, based on document, we allowed to define combined operation for batch. 
Even we can do this for now, it might need to be public if we are ready to do "delete|upload" batch operation.

Reply:
https://docs.microsoft.com/en-us/rest/api/searchservice/addupdate-or-delete-documents#response

Reply:
We can hide behind for now as we do not have batch operation on different action type. Can make it public when we provide the function.

Reply:
It cannot the private as we the index batch operation is supported.
`public Mono<IndexDocumentsResult> indexDocuments(IndexDocumentsBatch<?> batch)

Reply:
What we can do is to rename it to something make more sense